engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/Atom.java

```diff
@@ -183,6 +184,7 @@ Text toDisplayString(boolean allowSideEffects, @CachedLibrary("this") InteropLib
     try {
       return TypesGen.expectText(atoms.invokeMember(this, "to_text"));
     } catch (UnsupportedMessageException
+        | AbstractTruffleException
```

Comment:
This is a relatively easy fix. There already was a behavior that on (almost) any exception we fallback to `atom.toString(10)`. It even worked for Enso `Error` as it was caught by `UnexpectedResultException`. I am just expanding it to `Panic` and other guest languages exceptions.